docs/papers/architecture-whitepaper.md

```diff
@@ -0,0 +1,1709 @@
+- **Key Management Systems** - Software solutions that centralize key management tasks like generation, storage, access control, and lifecycle management.
+- **Key Encryption** - Encryption of cryptographic keys using master keys or key-encryption keys (KEKs) before storing them in persistent storage. Key encryption ensures that even if the storage medium is compromised, encrypted keys remain protected and cannot be used without decryption.
+
+###### Considerations
+- **Compliance and Governance** - Compliance with regulatory requirements and industry standards governing cryptographic key management, such as PCI DSS, GDPR, or FIPS 140-2, should be ensured by implementing robust security controls and audit mechanisms.
+- **Hierarchical Deterministic (HD) Keys** - enable many public/private key pairs to be generated from a single seed, such as a 12- or 24-word mnemonic phrase.
+- **Shamir’s Secret Sharing (SSS)** - Consider whether a secret sharing scheme where a key is split into multiple shares, and any pre-defined number of shares are required to reconstruct the original key should be utilized.
+
+##### Credential Storage
+**Credential Storage** refers to the ability to store and manage credentials. Credentials come in many different standardized formats (e.g., SD-JWT, W3C Verifiable Credentials, MDOC), as well as different encoding schemes (e.g., JSON, JSON-LD, CBOR). Credential Storage makes use of the [Data Storage](#data-storage) capability and sub-capabilities, as credentials are a specialized form of data. Specific capabilities needed for credential storage include status management and schema and definition retrieval.
+
+![Credential Storage](./images/credential-storage.png "Credential Storage")
+
+###### Sub-capabilities
+- **Status Management** - Ability to expose and manage the revocation status and expiration dates of the credential.
+- **Schema and Definition Retrieval** - Ability to retrieve the credential schemas and definitions from the verifiable data registry or the linked data.
+- **Encrypted Data Storage** - Ability to store the credentials in an encrypted format (see [Data Storage](#data-storage) capability for more information).
+
+###### Considerations
+- **Portability** Credential types necessitating cryptographic proof of ownership require a synchronized approach to credential portability and key recovery. This alignment is essential for scenarios like disaster recovery or transitioning to another device, as the holder must access their keys or secrets to present proof during verifier interactions. When keys for cryptographic proof of ownership reside in a hardware secure module and cannot be transferred to another device, credentials tied to those keys become non-functional on a different device, given the inability to establish proof of ownership.
```

Comment:
The consideration for portability is mentioned here but I would propose to give it more importance as governmental identity credentials heavily rely on device bound cryptographic keys.

Reply:
I will add something to reflect this.